filters/filters.go

```diff
@@ -0,0 +1,94 @@
+package filters
+
+import (
+	"net/http"
+
+	"github.com/getlantern/errors"
+
+	"github.com/getlantern/http-proxy/utils"
+)
+
+// Filter is like an http.Handler that
+type Filter interface {
+	// Apply is like the function on http.Handler but also gets a Next which
+	// allows it to continue execution along the current filter chain. If Apply
+	// returns an error, we will write an appropriate status code to the response.
+	// Tip - use filters.Fail() to provide an error with a description.
+	Apply(w http.ResponseWriter, req *http.Request, next Next) error
```

Comment:
I feel pretty good about this latest API.

1. It avoids confusion of the word "context"
2. It hides filters from each other, simply providing a Next function for executing
3. By providing the Next function it puts filters on their parents' call stack, which allows ops to work as designed
4. Relying on returned errors is convenient for short-circuiting interruption (i.e. if condition, return error, don't worry about code below here)
5. By providing the filters.Fail() convenience method, the return signature is clean but we still get to distinguish between a description of the error and the underlying cause (which is needed for returning the right HTTP status code)